psycop/common/model_training_v2/config/registries_testing_utils.py

```diff
@@ -0,0 +1,118 @@
+
+    filled_cfg.to_disk(base_filename.parent / out_filename)
+
+
+def generate_configs_from_registered_functions() -> bool:
+    """Generate config files from registered functions in the baseline registry
+    by filling with the default values. If no identical file exists on disk,
+    write the config to disk. This is useful for testing whether old configs
+    still work with the current codebase.
+    Returns bool indicating whether any new configs were written to disk."""
+    registries_dict = BaselineRegistry().to_dict()
+
+    generated_new_configs = False
+    for registry_name, registry in registries_dict.items():
+        for registered_function_name in registry.get_all():
+            cfg = Config(
+                {"placeholder": {f"@{registry_name}": f"{registered_function_name}"}},
+            )
+            try:
+                filled_cfg = BaselineRegistry().fill(cfg)
+            except ConfigValidationError as e:
+                if any(
+                    whitelisted_function in str(e)
+                    for whitelisted_function in REGISTERED_FUNCTION_WHITELIST
+                ):
+                    print(
+                        f"Encountered ConfigValidationError in {registered_function_name}, but whitelisted, skipping",
+                    )
+                    continue
+                if "*" in str(e):
+                    raise Exception(
+                        f"""Encountered ConfigValidationError in {registered_function_name}, with a function that's not whitelisted.
+        This means that the function has changed in a way that breaks backwards compatability, by adding a
+        *args argument. If this is intentional, add the function name to the whitelist in registries_testing_utils.py.
+        Otherwise, create a copy of the function before the change and update the name of the new function in the registry,
+        e.g. func_name_v2.
+        If you whitelist a function, make sure there is a config for it in the `static_registry_configs` folder.""",
+                    ) from e
+
+                raise Exception(
+                    f"""Encounted ConfigValidationError in {registered_function_name}. This means that the function has changed
+        in a way that breaks backwards compatability by e.g. adding a new, non-default argument.
+        Ensure that all arguments have default values and can be resolved by the registry.""",
+                ) from e
+
+            base_filename = (
+                STATIC_REGISTRY_CONFIG_DIR / registry_name / registered_function_name
+            )
+            ## check for identical files on disk
+            identical_file_exists = _identical_config_exists(
+                filled_cfg,
+                base_filename,
+            )
+            # if none, write to disk
+            if not identical_file_exists:
+                generated_new_configs = True
+                _save_cfg_to_disk(filled_cfg, base_filename)
```

Comment:
Just riffing, but as I see it, this will add a new .cfg if we change the value of the defaults as well. Are we sure we want that? Can we e.g. check if a file exists with the same keys instead?

Reply:
Or maybe this is fine.

Reply:
That's true -- don't think it's an issue though, as both configs should pass